ethcore/src/engines/mod.rs

```diff
@@ -83,12 +85,45 @@ pub enum EngineError {
 	RequiresClient,
 	/// Invalid engine specification or implementation.
 	InvalidEngine,
+	/// Requires signer ref, but none registered.
+	RequiresSigner,
+	/// Checkpoint is missing
+	CliqueMissingCheckpoint(H256),
+	/// Missing vanity data
+	CliqueMissingVanity,
+	/// Missing signature
+	CliqueMissingSignature,
+	/// Missing signers
+	CliqueCheckpointNoSigner,
+	/// List of signers is invalid
+	CliqueCheckpointInvalidSigners(usize),
+	/// Wrong author on a checkpoint
+	CliqueWrongAuthorCheckpoint(Mismatch<Address>),
+	/// Wrong checkpoint authors recovered
+	CliqueFaultyRecoveredSigners(Vec<String>),
+	/// Invalid nonce (should contain vote)
+	CliqueInvalidNonce(H64),
+	/// The signer signed a block to recently
+	CliqueTooRecentlySigned(Address),
+	/// Custom
+	Custom(String),
 }
 
 impl fmt::Display for EngineError {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
 		use self::EngineError::*;
 		let msg = match *self {
+			CliqueMissingCheckpoint(ref hash) => format!("Missing checkpoint block: {}", hash),
+			CliqueMissingVanity => format!("Extra data is missing vanity data"),
+			CliqueMissingSignature => format!("Extra data is missing signature"),
+			CliqueCheckpointInvalidSigners(len) => format!("Checkpoint block list was of length: {} of checkpoint but
+															it needs to be bigger than zero and a divisable by 20", len),
```

Comment:
Typo
```suggestion
															it needs to be bigger than zero and a divisible by 20", len),
```

Reply:
It looks like I didn't suggest anything, but you need to scroll to the write to see it :P